org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/Scope.java

```diff
@@ -5193,6 +5209,8 @@ private int parameterCompatibilityLevel(TypeBinding arg, TypeBinding param, Look
 		// only called if env.options.sourceLevel >= ClassFileConstants.JDK1_5
 		if (arg == null || param == null)
 			return NOT_COMPATIBLE;
+		if ((param.tagBits & TagBits.HasMissingType) != 0)
+			return NEEDS_MISSING_TYPE;
```

Comment:
NOT_DISCERNIBLE ??

Reply:
> NOT_DISCERNIBLE ??

pardon?

Reply:
I was trying to come up with an alternate name for the new constant - Not happy with `NEEDS_MISSING_TYPE`. Not happy with NOT_DISCERNIBLE either actually. COMPATIBLE_BUTFOR_MISSING_TYPE is not entirely satisfactory either ... but is may be better ?

Reply:
> Some more to pick from:

> * COMPATIBLE_IGNORING_MISSING_TYPE?

This looks very apt. +1